File: dotnet/Endpoints/Mappers/TodoDtoMappers.cs
using Todo.Core.Models;
using Todo.Endpoints.Models;

namespace Todo.Endpoints.Mappers;

internal static class TodoDtoMappers
{
    public static TodoDto ToDto(this TodoEntity entity) => new(
        Id: entity.Id,
        UpdatedAt: entity.UpdatedAt,
        DeadlineAt: entity.DeadlineAt,
        Title: entity.Title,
        Description: entity.Description,
        IsDone: entity.IsDone
    );

    public static ICollection<TodoDto> ToDtos(this IEnumerable<TodoEntity> entities)
        => entities.Select(ToDto).ToArray();
}

File: nodejs/src/endpoints/mappers/todo-dto.mappers.ts
import { Todo } from "../../core/todo";
import { TodoDto } from "../models/todo.dto";

export const ToTodoDto = (todo: Todo) =>
  <TodoDto>{
    id: todo.id,
    updatedAt: todo.updatedAt,
    deadlineAt: todo.deadlineAt,
    title: todo.title,
    description: todo.description,
    isDone: todo.isDone,
  };

export const ToTodoDtoList = (todos: Todo[]): TodoDto[] => todos.map(ToTodoDto);
